include/astshim/ChebyMap.h

```diff
@@ -0,0 +1,321 @@
+
+    /**
+    This function creates a new @ref ChebyMap which is a copy of this one,
+    in which a specified transformation (forward or inverse)
+    has been replaced by a new Chebyshev polynomial transformation. The
+    coefficients of the new transformation are estimated by sampling
+    the other transformation and performing a least squares polynomial
+    fit in the opposite direction to the sampled positions and values.
+
+    This method can only be used on (1-input,1-output) or (2-input, 2-output)
+    @ref ChebyMap "ChebyMaps".
+
+    The transformation to create is specified by the `forward` parameter.
+    In what follows "X" refers to the inputs of the @ref ChebyMap, and "Y" to
+    the outputs of the @ref ChebyMap. The forward transformation transforms
+    input values (X) into output values (Y), and the inverse transformation
+    transforms output values (Y) into input values (X). Within a @ref ChebyMap,
+    each transformation is represented by an independent set of
+    polynomials, P_f or P_i: Y=P_f(X) for the forward transformation and
+    X=P_i(Y) for the inverse transformation.
+
+    The `forward` parameter specifies the transformation to be replaced.
+    If it is false, a new forward transformation is created
+    by first finding the input values (X) using the inverse transformation
+    (which must be available) at a regular grid of points (Y) covering a
+    rectangular region of the @ref ChebyMap's output space. The coefficients of
+    the required forward polynomial, Y=P_f(X), are chosen in order to
+    minimise the sum of the squared residuals between the sampled values
+    of Y and P_f(X).
+
+    If `forward` is false (probably the most likely case),
+    a new inverse transformation is created by
+    first finding the output values (Y) using the forward transformation
+    (which must be available) at a regular grid of points (X) covering a
+    rectangular region of the @ref ChebyMap's input space. The coefficients of
+    the required inverse polynomial, X=P_i(Y), are chosen in order to
+    minimise the sum of the squared residuals between the sampled values
+    of X and P_i(Y).
+
+    This fitting process is performed repeatedly with increasing
+    polynomial orders (starting with linear) until the target
+    accuracy is achieved, or a specified maximum order is reached. If
+    the target accuracy cannot be achieved even with this maximum-order
+    polynomial, the best fitting maximum-order polynomial is returned so
+    long as its accuracy is better than
+    "maxacc". If it is not, an error is reported.
+
+    @param[in] forward  If true the forward transformation is replaced.
+                    Otherwise the inverse transformation is replaced.
+    @param[in] acc  The target accuracy, expressed as a geodesic distance within
+                    the ChebyMap's input space (if `forward` is false)
+                    or output space (if `forward` is true).
```

Comment:
False is the standard case, so I prefer to mention it first. The docs some from AST but you are right they could be better (and AST uses an int for `forward` so I had to change this slightly).

I had the docs wrong for awhile -- I thought the domain applied to the function being generated, but it applies to the function that is being evaluated in order to generate its inverse. (Though I knew better when I wrote the PolyTran unit test). And I had false/true swapped in the docs as a result.